pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1004,6 +1052,57 @@ BPF_PERF_OUTPUT(net_events, 1024);  // network events submission
 
 static __always_inline void *get_path_str(struct path *path);
 
+// HELPERS: ERRORS ---------------------------------------------------------------------------------
+
+static __always_inline void do_tracee_error(
+    void *ctx, enum bpf_error_level level, enum bpf_error_id id, s64 ret, u32 line, void *file)
+{
+    if (!ctx || !file)
+        return;
+
+    u32 zero = 0;
+    bpf_error_output_t *err_output = bpf_map_lookup_elem(&error_output_scratch, &zero);
+    if (unlikely(err_output == NULL))
+        return;
+
+    err_output->level = level;
+    err_output->id = id;
+
+    err_output->err.ret = ret;
+    err_output->err.cpu = bpf_get_smp_processor_id();
+    err_output->err.line = line;
+    u64 fsize = __builtin_strlen(file);
+    if (unlikely(fsize >= BPF_MAX_ERR_FILE_LEN))
+        fsize = BPF_MAX_ERR_FILE_LEN - 1;
+    __builtin_memcpy(err_output->err.file, file, fsize);
+    err_output->err.file[fsize] = '\0';
+
+    bpf_error_count_t counter_buf = {};
+    counter_buf.count = 1;
+    counter_buf.ts = bpf_ktime_get_ns(); // store the current ts
+    u64 ts_prev = 0;
+
+    bpf_error_count_t *counter = bpf_map_lookup_elem(&errors_count, &err_output->err);
+    if (likely(counter != NULL)) {
+        ts_prev = counter->ts; // store previous ts
+
+        counter->count += 1;
```

Comment:
you're changing a pointee value on the map without an atomic operation. I believe this is racy among different CPUs (as the "errors_count" map isn't a per CPU map). So, you might need either to copy the counter to the stack, increment and update the key with the new stack values (the key update is atomic) OR use atomic_xchg() for the pointer (not sure if that works for a map pointer, need to check).

Reply:
I believe, due to the nature of this counter, there is no problem if we loose a few increments (due to 2 cpus copying the values to the stack at the same time and both updating the key values with their stack values). With that, it would be just a question of memcping the pointer to the stack, incrementing and updating the key.

Reply:
Not doing that might get us into a situation where count is in diff cachelines, and it gets badly corrupted, not that would impact too much, but, still...

Reply:
https://github.com/geyslan/tracee/pull/7

Reply:
Geyslan made a good point that part of the key to this map is the cpu id.
In that case, it is not possible that two different CPUs will update the same counter

Reply:
Indeed it is a good point (haven't payed attention to it). Nice work @geyslan !!